src/modules/commander/Arming/PreFlightCheck/checks/sdcardCheck.cpp

```diff
@@ -43,30 +44,29 @@
 
 bool PreFlightCheck::sdcardCheck(orb_advert_t *mavlink_log_pub, const bool report_fail)
 {
-	const char *microsd_dir = PX4_STORAGEDIR;
 	bool success = true;
-	struct statfs statfs_buf;
-	uint64_t total_bytes = 0;
-
-	int32_t reporting_enabled = 0;
-	param_get(param_find("COM_ARM_SDCARD"), &reporting_enabled);
-
-	if (reporting_enabled == 0) {
-		return success;
-	}
 
-	if (statfs(microsd_dir, &statfs_buf) == 0) {
-		total_bytes = (uint64_t)statfs_buf.f_blocks * statfs_buf.f_bsize;
-	}
+	int32_t param_com_arm_sdcard{0};
+	param_get(param_find("COM_ARM_SDCARD"), &param_com_arm_sdcard);
 
-	if (total_bytes == 0) { // on NuttX we get 0 total bytes if no SD card is inserted
+	if (param_com_arm_sdcard > 0) {
+		bool sdcard_present{false};
+		struct statfs statfs_buf;
 
-		if (reporting_enabled == 2) {
-			success = false;
+		if (statfs(PX4_STORAGEDIR, &statfs_buf) == 0) {
+			// on NuttX we get 0 total bytes = f_blocks * f_bsize if no SD card is inserted
```

Comment:
The comment has to be adapted

Reply:
I did but probably I should make it unambiguous.

Reply:
I made it more clear, what do you think?